jormungandr/src/diagnostic.rs

```diff
@@ -0,0 +1,73 @@
+use crate::start_up::Error;
+use std::fmt::{self, Display, Formatter};
+
+#[cfg(unix)]
+pub type DiagnosticError = nix::Error;
+
+#[cfg(not(unix))]
+custom_error! {pub DiagnosticError {} = "diagnostic error"}
```

Comment:
Please convert this to a derive with `thiserror`. Let's bring this error stuff to a single good standard.

Reply:
Oops, just missed that we already have it in the project. Btw, why not the `failure` crate?